src/webhook/test_regular_user_validation.py

```diff
@@ -1,23 +1,70 @@
 import unittest
+import json
 
 from webhook.regular_user_validation import is_request_allowed
+from webhook.regular_user_validation import get_response
 
 ADMIN_GROUPS = [
     "osd-sre-admins"
 ]
 
+def create_request(uid, kind, username, groupname):
+    request = {
+        "json": {
+            "request": {
+                "uid": "testuser",
+                "operation": "UPDATE",
+                "userInfo": {
+                    "username": username
+                },
+                "oldObject": {
+                    "metadata": {
+                        "name": groupname
+                    }
+                },
+                "object": {
+                    "kind": kind,
+                    "metadata": {
+                        "name": groupname
+                    }
+                },
+            }
+        }
+    }
+    return request
```

Comment:
Following on from above, I think you could avoid having to change the way you're accessing `request.json` by making this:

```suggestion
    class FakeRequest(object):
        json = {
            "request": {
                "uid": "testuser",
                "operation": "UPDATE",
                "userInfo": {
                    "username": username
                },
                "oldObject": {
                    "metadata": {
                        "name": groupname
                    }
                },
                "object": {
                    "kind": kind,
                    "metadata": {
                        "name": groupname
                    }
                },
            }
        }

    return FakeRequest()
```